stix_shifter_modules/infoblox/stix_transmission/api_client.py

```diff
@@ -0,0 +1,217 @@
+import json
+
+from stix_shifter_utils.stix_transmission.utils.RestApiClient import RestApiClient
+from stix_shifter_utils.utils import logger
+
+_USER_AGENT = 'IBV1StixShifter/1.0'
+_MAX_RESULT = 10000
+
+class APIClient:
+
+    def __init__(self, connection, configuration):
+        self.logger = logger.set_logger(__name__)
+        self.endpoint_start = ''
+        headers = dict()
+        host_port = connection.get('host') + ':' + str(connection.get('port', ''))
+        headers['accept'] = 'application/json'
+        auth = configuration.get('auth')
+        if auth is not None and auth.get('token', None) is not None:
+            headers['Authorization'] = 'token {}'.format(auth.get('token'))
+        url_modifier_function = None
+        headers['user-agent'] = _USER_AGENT
+
+        self.timeout = connection['options'].get('timeout')
+        self.result_limit = connection['options'].get('result_limit')
+        if self.result_limit > _MAX_RESULT:
+            self.logger.warning("The length exceeds length limit. Use default length: %s", _MAX_RESULT)
+            self.result_limit = _MAX_RESULT
+
+        self.client = RestApiClient(host_port,
+                                    None,
+                                    headers,
+                                    url_modifier_function,
+                                    cert_verify=connection.get('selfSignedCert', True),
+                                    sni=connection.get('sni', None)
+                                    )
+
+    def ping_data_source(self):
+        # Pings the data source
+        endpoint = 'tide/api/data/threats/state'
+        # now = datetime.datetime.utcnow().isoformat(timespec="milliseconds") + "Z"
+        # https://csp.infoblox.com:443/tide/api/data/threats/state?type=host&rlimit=1
+
+        resp = self.client.call_api(endpoint, 'GET', timeout=self.timeout, urldata={"type": "host", "rlimit": "1"})
+        return {"code": resp.code}
```

Comment:
I would return the whole response instead of just the error code to have a better explanation of the error.

`return self.client.call_api(endpoint, 'GET', timeout=self.timeout, urldata={"type": "host", "rlimit": "1"})`

Reply:
Updated, here is an example error response:
```
❯ python3 main.py transmit infoblox '{"host":"csp.infoblox.com"}' '{"auth": {"token":"xxx"} }' ping
{
    "success": false,
    "error": "Invalid type --- type [host, ip, url, hash, email] must be specified in path or parameters\n",
    "code": "invalid_parameter"
}
```